pandas/core/arrays/array_.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+
+from pandas._libs import lib, tslibs
+
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+from pandas import compat
+
+
+def array(data, dtype=None, copy=False):
+    """
+    Create an array.
+
+    .. versionadded:: 0.24.0
+
+    Parameters
+    ----------
+    data : Sequence[object]
+        A sequence of objects. The scalars inside `data` should
+        be instances of the scalar type for `dtype`.
+
+        When `data` is an Index or Series, the underlying array
+        will be extracted from `data`.
+
+    dtype : Union[str, np.dtype, ExtensionDtype], optional
```

Comment:
Do we want to use such type annotation terminology in user facing docstrings? I am fine with having them already as type comments, but using them in docstrings in general requires some discussion I think (to say, it's not in the current docstring guidelines). 
I think not everybody will now what we mean here with "Union"

Reply:
I think this is pretty common nowadays and pretty useful information.

Reply:
For people who don't see themselves as programmers (=most of our users) and who only interact with numpy, pandas, matplotlib, ... docstrings, I don't think this is "common"

(and it is also not really giving extra information, as "str, np.dtype or ExtensionDtype" also says "one of those three".

Reply:
All change to prose for now, but I think we'll want to do some educating of users here.